Code/Core/FileIO/FileIO.cpp

```diff
@@ -785,6 +803,14 @@
         FindClose( hFind );
 
     #elif defined( __LINUX__ ) || defined( __APPLE__ )
+        // Special case symlinks.
+        struct stat stat_source;
+        VERIFY( lstat( pathCopy.Get(), &stat_source ) == 0 );
+        if ( S_ISLNK( stat_source.st_mode ) )
+        {
+            return;
```

Comment:
Per the Windows behavior, `FindFileEx()` won't traverse a symlink, where as I'm 99% sure `opendir()` will, so we want to early out in that case for consistent behavior.